src/accel/AlongStepFactory.hh

```diff
@@ -135,6 +144,15 @@ class RZMapFieldAlongStepFactory : public AlongStepFactoryInterface
     // Construct with a function to return RZMapFieldInput
     explicit RZMapFieldAlongStepFactory(RZMapFieldFunction f);
 
+    //!@{
+    //! Intent to use with \c std::function --> requires CopyConstructible
+    RZMapFieldAlongStepFactory(RZMapFieldAlongStepFactory const&);
+    RZMapFieldAlongStepFactory& operator=(RZMapFieldAlongStepFactory const&);
+    RZMapFieldAlongStepFactory(RZMapFieldAlongStepFactory&&) = default;
+    RZMapFieldAlongStepFactory& operator=(RZMapFieldAlongStepFactory&&)
+        = default;
+    //!@}
+
```

Comment:
@pcanal Do the changes to this file really seem worth it to you? This was an interesting experiment but adding 60 lines of code to prevent slicing is just too much. This change makes the code a *little* more foolproof, but *way* harder to understand for everyone.

Reply:
In that case, it's mostly `Interface` makes the subclass non-copy-onstructible but the purpose of `*AlongStepFactory` is to be used to construct a `std::function` which required its argument to be copy-constructible..

Reply:
Only the abstract class needs to be unable to copy/move construct; a `final` daughter class should act exactly the same as a regular object.

Reply:
60 lines is probably too much but it can be smaller and clearer: https://github.com/celeritas-project/celeritas/pull/834/files#r1246958488